src/devices/wifi/nm-device-iwd.c

```diff
@@ -864,6 +869,30 @@ check_scanning_prohibited (NMDeviceIwd *self, gboolean periodic)
 	return prohibited;
 }
 
+static void
+scan_cb (GObject *source, GAsyncResult *res, gpointer user_data)
+{
+	NMDeviceIwd *self = user_data;
+	NMDeviceIwdPrivate *priv = NM_DEVICE_IWD_GET_PRIVATE (self);
+	NMDeviceState state = nm_device_get_state (NM_DEVICE (self));
+	gs_free_error GError *error = NULL;
+
+	if (   !_nm_dbus_proxy_call_finish (G_DBUS_PROXY (source), res,
+	                                   G_VARIANT_TYPE ("()"), &error)
+	    && g_error_matches (error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
```

Comment:
`scan_cb` must not dereference `scan` before checking that the request was not cancelled. Otherwise, `self` might be a dangling pointer, and `NM_DEVICE_IWD_GET_PRIVATE (self);` and `nm_device_get_state (NM_DEVICE (self));` will lead to crash.

Reply:
Oops